networking/src/NTP/Client.hs

```diff
@@ -280,22 +277,19 @@ stopNtpClient cli = do
     -- unblock receiving from socket in case no one replies
     forM_ sockets $ \s -> liftIO (close s) `catchAny` (const $ pure ())
 
-startNtpClient :: NtpMonad m => NtpClientSettings m -> m (NtpStopButton m)
-startNtpClient settings =
+withNtpClient :: NtpMonad m => NtpClientSettings m -> (NtpStopButton m -> m a) -> m a
+withNtpClient settings callback =
     withSocketsDoLifted $
     bracketOnError (mkSockets settings) closeSockets $ \sock -> do
         cli <- mkNtpClient settings sock
 
         addrs <- catMaybes <$> mapM (resolveHost cli $ socketsToBoolDescr sock)
                                     (ntpServers settings)
-        if null addrs then
-            throwM NoHostResolved
-        else do
-            void . fork . withSocketsDoLifted $ startReceive cli
-            void . fork . withSocketsDoLifted $ startSend addrs cli
-            log cli Info "Launched"
-
-        return $ NtpStopButton $ stopNtpClient cli
+        when (null addrs) $ throwM NoHostResolved
+        withAsync (withSocketsDoLifted $ startReceive cli) $ \_ ->
```

Comment:
`withSocketsDo` uses global counter to track whether Windows socket library should be attached, so one `withSocketsDoLifted` in the beginning would be enough

Reply:
> `withSocketsDo` uses global counter to track whether Windows socket library should be attached, so one `withSocketsDoLifted` in the beginning would be enough

As you can see, old code was using it more than once.
```
            void . fork . withSocketsDoLifted $ startReceive cli
            void . fork . withSocketsDoLifted $ startSend addrs cli
```

So I decided to preserve old behaviour. Do I understand correctly that in old code usage of these `withSocketsDoLifted` was redundant as well?

Reply:
In old code you could not know for sure (at least at first sight) which of the threads will finish earlier, and both require socket library attached, so both are wrapped into `withSocketsDo`.
Now you can happily use only single `withSocketsDo` which wraps over all threads.